lib/ansible/modules/lineinfile.py

```diff
@@ -542,9 +585,10 @@ def main():
     path = params['path']
     firstmatch = params['firstmatch']
     regexp = params['regexp']
+    search_string = params['search_string']
     line = params['line']
 
-    if regexp == '':
+    if regexp == '' or search_string == '':
         module.warn(
             "The regular expression is an empty string, which will match every line in the file. "
             "This may have unintended consequences, such as replacing the last line in the file rather than appending. "
```

Comment:
Adjust the warning message so it is appropriate.
```suggestion
    if '' in [regexp, search_string]:
        msg = ("The %s is an empty string, which will match every line in the file. "
               "This may have unintended consequences, such as replacing the last line in the file rather than appending.")
        param_name = 'search_string'
        if regexp == '':
            param_name = 'regular expression'
            msg += " If this is desired, use '^' to match every line in the file and avoid this warning."

        module.warn(msg % param_name)
```

Reply:
❤️